packages/playwright/src/AxeCoreWrapper/DefaultAxeCoreWrapper.cs

```diff
@@ -1,108 +1,297 @@
+                    if (frameHandle != null)
+                    {
+                        var childFrameElement = frameHandle.AsElement();
+                        if (childFrameElement == null)
+                        {
+                            partialResults.Add(null);
+                            continue;
+                        }
+
+                        var childFrame = await childFrameElement.ContentFrameAsync();
+                        if (childFrame == null)
+                        {
+                            partialResults.Add(null);
+                            continue;
+                        }
+
+                        partialResults.AddRange(await RunPartialRecursive(childFrame, options, frameContext, false, iframes));
+                    }
+                    else
+                    {
+                        partialResults.Add(null);
+                    }
+                }
+                catch (Exception)
+                {
+                    partialResults.Add(null);
+                }
+            }
+
+            return partialResults;
+        }
+
+        private async Task<AxeResult> IsolatedFinishRun(IFrame frame, object?[] partialResults, object options)
+        {
+            var browser = frame.Page.Context.Browser;
+            if (browser == null)
+            {
+                throw new Exception("Could not locate browser associated with page");
+            }
+            var blankPage = await browser.NewPageAsync();
+            try
+            {
+                await ConfigureAxe(blankPage.MainFrame);
+                var serializedPartials = JsonConvert.SerializeObject(partialResults, JsonSerializerSettingsFinishRun);
+                var result = await blankPage.EvaluateAsync<object>(EmbeddedResourceProvider.ReadEmbeddedFile("finishRun.js"), new[] { serializedPartials, options });
+                return DeserializeResult(result);
+            }
+            finally
+            {
+                await blankPage.CloseAsync();
+            }
+        }
+
+        /// <summary>
+        /// Execute `axe.utils.getFrameContexts(context)`
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        private async Task<List<AxeFrameContext>> GetFrameContexts(IFrame frame, object context)
+        {
+            var frameContextsString = await frame.EvaluateAsync<string>(EmbeddedResourceProvider.ReadEmbeddedFile("getFrameContexts.js"), context);
+            var frameContexts = JsonConvert.DeserializeObject<List<AxeFrameContext>>(frameContextsString);
+            if (frameContexts == null)
+            {
+                return new List<AxeFrameContext>();
+            }
+            return frameContexts;
+        }
+
+        /// <summary>
+        /// Injects scripts into the current frame and configures axe
+        /// </summary>
+        private async Task ConfigureAxe(IFrame frame)

```

Comment:
The `RunLegacy` paths still refer to `EmbedAxeCoreIntoPage` directly rather than going through this method that includes setting branding. Should they be updated to go through this path?

Reply:
`EmbedAxeCoreIntoPage` was updated to set branding.

Reply:
Ah, my mistake, you're right - I got confused by `EmbedAxeCoreIntoFrame` and `EmbedAxeCoreIntoPage` having different behavior around whether they included the branding step. Would it make sense to have them be consistent, since the names are so similar?